test/int/databaseuser_unprotected_test.go

```diff
@@ -241,9 +241,11 @@ var _ = Describe("Atlas Database User", Label("int", "AtlasDatabaseUser", "prote
 
 			By("Giving user readWrite permissions", func() {
 				// Adding the role allowing read/write
-				testDBUser1 = testDBUser1.WithRole("readWriteAnyDatabase", "admin", "")
-
-				Expect(k8sClient.Update(context.Background(), testDBUser1)).To(Succeed())
+				Eventually(func(g Gomega) {
```

Comment:
Are these (this and the next ones) a good candidate to use the `retryUpdateOnConflict` added here https://github.com/mongodb/mongodb-atlas-kubernetes/pull/1622

Reply:
pontentially, let me first rebase this one first

Reply:
actually, I decided to remove the deflaking of database users out of this PR, it is a separate contribution.